### ISSUE TITLE
Permissions of the file not preserved in zip

### ISSUE THREAD

**Issue by mruz** [terralist/terralist#213]
I have a bash script in a module with execute permissions. 
After I upload it to the registry, script looses that permission. 
Then when I try to use the module I'm getting:
> Error: External Program Execution Failed
Error Message: /bin/bash: .terraform/modules/my_module/script.sh: Permission denied

Do we need to read fileinfo os.Stat() and pass it to zip.FileInfoHeader(fileinfo) at  [inmemory](https://github.com/terralist/terralist/blob/master/pkg/file/inmemory.go#L24)?

**Comment by valentindeaconu** [comment]
Hello! Thanks for reporting this.

I already iterate over the files while building the `InMemoryFile` array:

https://github.com/terralist/terralist/blob/master/pkg/file/fetch.go#L151

So, I guess, fixing this issue is a matter of adding an extra `fs.FileInfo` member in the `InMemoryFile` struct and pass it from the Walk method to the archive builder via the `FileInfoHeader` method you suggested.

**Comment by mruz** [comment]
If I download .zip from github I can see the permissions, but they're missing on archive from registry
```
$ zipinfo github-modules-my_module-v1.0.0.zip | grep script.sh
-rwxr-xr-x  2.3 unx     1341 tx defN 23-Jul-24 14:54 github-modules-my_module-v1.0.0/my_module/script.sh

$ zipinfo terraform-registry-1.0.0.zip | grep script.sh
-rw----     2.0 fat     1341 bl defN 80-000-00 00:00 script.sh
```

It would be nice to add `archive/tar` support as well.

**Comment by valentindeaconu** [comment]
Closing this since it was fixed in #215 and released in v0.4.7.
